src/app.js

```diff
@@ -1 +1,49 @@
-'use strict';
+/* eslint-disable no-console */
+"use strict";
+
+const readline = require("readline");
+const fs = require("fs/promises");
+
+const terminal = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+let userFilename = "";
+
+function controller(flag, value) {
+  if (flag === "filename") {
+    getFileData(value);
```

Comment:
The function 'getFileData' is called with the 'value' parameter which could potentially be undefined or empty. Please ensure that a validation or error handling is in place in the 'getFileData' function to handle such cases.